File: leetcode/Balanced Binary Tree.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return a boolean
    def isBalanced(self, root):
        
        if root == None:
            return True
        left = self.depth(root.left)
        right = self.depth(root.right)
        return abs(left-right) <= 1 and self.isBalanced(root.left) and self.isBalanced(root.right)
    
    def depth(self, root):
        if root == None:
            return 0
        return max(self.depth(root.left), self.depth(root.right)) + 1

File: leetcode/cpp/Balanced_Binary_Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isBalanced(TreeNode* root) {
        if (height(root) == -1)
            return false;
        else
            return true;
    }
    
    int height(TreeNode * root) {
        if (NULL == root)
            return 0;
        int left = height(root->left);
        int right = height(root->right);
        if (left == -1 || right == -1)
            return -1;
        if (abs(left - right) > 1)
            return -1;
        return max(left, right) + 1;
    }
};